### PR TITLE
Fix: remove long message text and scrollview from message overlay [CRNS - 278]

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal
This PR introduces the fix for the long message scroll problem due to message overlay text and the related ScrollView problem.

## 🛠 Implementation details
Previously, opening the message overlay on long-press loaded a Scrollview which generated a warning/error of Virtualized list due to the presence of Reaction list within the view. To solve this the long text size is truncated and the underlying Scrollview is removed.

I have introduced a prop `messageTextNumberOfLines` to solve the issue which gives users flexibility to change the prop value later and render the number of lines according to their own purpose/use-case. By default, the value is 5.

Reference - https://stream-io.atlassian.net/browse/CRNS-278

## 🎨 UI Changes

<details>
    <summary>iOS</summary>

<table>
    <thead>
        <tr><td>Before</td><td>After</td></tr>
    </thead>
    <body><tr><td> </td><td></td></tr></tbody>
</table>

</details>

<details>
    <summary>Android</summary>

| Before | After |
| --- | --- |
| Screenshot 2021-11-01 at 21 33 09 | Screenshot 2021-11-01 at 21 33 27 |

</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [x] Screenshots added for visual changes
- [x] Documentation is updated

### CODE REVIEW

File: package/src/components/Message/MessageSimple/utils/renderText.tsx [hunk 1]
```diff
@@ -198,6 +201,20 @@ export const renderText = <
     return link;
   };
 
+  const paragraphText: ReactNodeOutput = (node, output, { ...state }) => {
+    const paragraph = React.createElement(
```
[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx:205]: JSX should work as well I guess, its worth updating :)

[the remark this answers is not in the document]
Comment (khushal87) [reply package/src/components/Message/MessageSimple/utils/renderText.tsx]: Done @vishalnarkhede 😄

File: package/src/components/MessageOverlay/MessageOverlay.tsx [hunk 2]
```diff
@@ -364,215 +365,203 @@ const MessageOverlayWithContext = <
...
+                                    (!hasThreadReplies || threadList)
+                                      ? borderRadiusS
+                                      : borderRadiusL,
+                                  borderBottomRightRadius:
+                                    (groupStyle === 'right_bottom' ||
+                                      groupStyle === 'right_single') &&
+                                    (!hasThreadReplies || threadList)
+                                      ? borderRadiusS
+                                      : borderRadiusL,
+                                  borderColor: grey_whisper,
```
[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply package/src/components/MessageOverlay/MessageOverlay.tsx:443]: This was one of the reason why we added these big ternaries in codebase - https://github.com/GetStream/stream-chat-react-native/pull/532/files#r592662391

But I agree that its extreamly hard to read this, so probably best to get rid of it.

[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply package/src/components/MessageOverlay/MessageOverlay.tsx:443]: Lets discuss in our team meeting on wednesday

**Review by vishalnarkhede — APPROVED** [review]
LGTM

**Comment by vishalnarkhede** [comment]
@khushal87 those screenshots in description are not rendered for some reason. Do you mind fixing that?

**Pull request merged**